qiskit/providers/ibmq/random/ibmqrandomservice.py

```diff
@@ -110,7 +110,8 @@ def __dir__(self) -> Dict:
         return self.__dict__
 
     def __getattr__(self, item: str) -> Any:
-        self._discover_services()
+        if not self.__getattribute__('_initialized'):
+            self._discover_services()
```

Comment:
This still checks `if not self._initialized:` which will call this code and put you in an infinite loop won't it? Why not just set self._initialized = False in the __init__ method?

Reply:
`self._initialized` **is** set in `__init__()`. The problem is `_initialized` is not present yet when `__getattr__()` is called to find `__getstate__` to pickle the instance to be passed to another process. 

If `_initialized` is a known attribute (i.e. `self.__getattribute__('_initialized')` would return True), then the call `self._initialized` will be directed to `__getattribute__()`, not `__getattr__()`.